server/src/main/proto/android_attestation.proto

```diff
@@ -0,0 +1,386 @@
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/wrappers.proto";
+import "google/type/date.proto";
+
+option java_package = "com.google.android.attestation";
+
+message KeyDescription {
+  // The key attestation format version
+  .google.protobuf.UInt32Value attestation_version = 1;
+
+  // This data structure indicates the extent to which a software feature, such
+  // as a key pair, is protected based on its location within the device.
+  // Note: The numeric values do not match the ASN.1 representation.
+  enum SecurityLevel {
+    SECURITY_LEVEL_UNSPECIFIED = 0;
+    // The logic for creating and managing the feature is implemented in the
+    // Android system. For the purposes of creating and storing key pairs, this
+    // location is less secure than the TEE but is more secure than your app's
+    // process space.
+    SOFTWARE = 1;
+    // The logic for creating and managing the feature is implemented in secure
+    // hardware, such as a TEE. For the purposes of creating and storing key
+    // pairs, this location is more secure because secure hardware is highly
+    // resistant to remote compromise.
+    TRUSTED_ENVIRONMENT = 2;
+    // The logic for creating and managing the feature is implemented in a
+    // dedicated hardware security module. For the purposes of creating and
+    // storing key pairs, this location is more secure because it is highly
+    // resistant to remote compromise and hardware attacks against the module.
+    // Present only in key attestation version ATTESTATION_V_THREE and later.
+    STRONG_BOX = 3;
+  }
+
+  // The security level of the attestation.
+  SecurityLevel attestation_security_level = 2;
+
+  // The version of the KeyMint (or Keymaster) implementation
+  .google.protobuf.UInt32Value key_mint_version = 3;
+
+  // The security level of the KeyMint implementation.
+  SecurityLevel key_mint_security_level = 4;
+
+  bytes attestation_challenge = 5;
+
+  // This value identifies the device—but only for a limited period of time. It
+  // is computed and is only used by system apps. In all other apps, uniqueId is
+  // empty.
+  bytes unique_id = 6;
+
+  // This data structure contains the key pair's properties themselves, as
+  // defined in the KeyMint hardware abstraction layer (HAL). You compare
+  // these values to the device's current state or to a set of expected values
+  // to verify that a key pair is still valid for use in your app.
+  message AuthorizationList {
+    // Purposes for which the key may be used.
+    enum KeyPurpose {
+      ENCRYPT = 0;
+      DECRYPT = 1;
+      SIGN = 2;
+      VERIFY = 3;
+      /* 4 is reserved */
+      WRAP_KEY = 5;
+      AGREE_KEY = 6;
+      ATTEST_KEY = 7;
+    }
+
+    repeated KeyPurpose purposes = 1;
+
+    // Specifies the cryptographic algorithm with which the key is used.
+    enum Algorithm {
+      ALGORITHM_UNSPECIFIED = 0;
+      RSA = 1;
+      EC = 3;
+      AES = 32;
+      TRIPLE_DES = 33;
```

Comment:
Why 33 for Triple DES?

Reply:
That's the value from Algorithm.aidl.  Asymmetric algorithms are 1-31, block ciphers 32-127, MAC algorithms 128+.